src/expr/src/vector_op/cast.rs

```diff
@@ -80,6 +80,14 @@ fn parse_naive_datetime(s: &str) -> Result<NaiveDateTime> {
     }
 }
 
+#[inline]
+pub fn i64_to_timestamp(ms: i64) -> Result<NaiveDateTimeWrapper> {
+    Ok(NaiveDateTimeWrapper::new(NaiveDateTime::from_timestamp(
+        ms / 1000,
+        (ms % 1000) as u32 * 1_000_000,
+    )))
+}
```

Comment:
This assumes the input is always in milliseconds, but some sources may use a different unit (e.g. microsecond in Debezium). Had a sync with @tabVersion recently and we decided to apply the following heuristic from epochconverter.com:
* [0, 1e11) are assumed to be in seconds
* [1e11, 1e14) are assumed to be in milliseconds
* [1e14, 1e17) are assumed to be in microseconds
* [1e17, upper) are assumed to be in nanoseconds

This would cause no problem for timestamp in [1973-03-03 09:46:40, 5138-11-16 09:46:40), which is practical for a streaming source.

Reply:
Do we need to allow users to use per-column annotations after providing this heuristic solution?

Reply:
It adds some complexity but little practical value. So I guess only when it is really needed.